### DISCUSSION TITLE
Callback URL Timeout Error

### DISCUSSION THREAD

**Discussion by art804** [jbergler/hass-ttlock#96] [Q&A] [answered]
## Describe the bug

I have set up the integration and the TTLock API. I receive the callback URL as a HA system notification. However, this URL is a local URL (http://10.1.1.10:8123/api/webhook/longstringoftextandnumbers). When I enter this URL on the TTLock website as the Callback URL I get a Java Socket timeout error.

HA is set up in a Docker container and I don't have external web access set up.

What am I doing wrong?

### To Reproduce

As above

### Expected behavior

Not to receive an error. I'm sure I'm missing something basic but the GitHub readme doesn't specify what the callback URL is supposed to look like.

## Diagnostic info

N/A, issue with third party site

**Comment by jbergler** [comment, marked as answer]
The callback url needs to be accessible from the internet.
When you lock/unlock a door, the webhook comes from the ttlock cloud rather than from the local gateway.

**Reply by art804** [reply]
Thank you, that makes sense and I have now got the Callback URL working on the TTLock website.
I found this post that explained it: https://github.com/jbergler/hass-ttlock/issues/1#issuecomment-1819735891

**Reply by CymruDad78** [reply]
> The callback url needs to be accessible from the internet. When you lock/unlock a door, the webhook comes from the ttlock cloud rather than from the local gateway.

Hi jbergler, I have just set up the intergration today and got it working successfully. But I don’t know how to set-up the callback url on the euopen.ttlock platform. 

Can you tell me where I find the URL please? I’m a little confused with this part of the set-up.

Thanks 🙏

**Comment by mremedi2023** [comment]
Should I create a rule in my router so that the application can access from outside? That would leave that port open, is that correct?
